.github/workflows/tiiuae-pixhawk-and-saluki.yaml

```diff
@@ -9,276 +9,33 @@ on:
   workflow_dispatch:
 
 jobs:
-  pixhawk:
-    name: build pixhawk
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout px4-firmware
-        uses: actions/checkout@v3
-        with:
-          path: px4-firmware
-          fetch-depth: 0
-      - name: Run px4-firmware pixhawk build
-        run: |
-          set -eux
-          mkdir -p bin
-          cd px4-firmware/
-          ./clone_public.sh
-          ./build.sh ../bin/ pixhawk
-          ls ../bin
-      - name: Upload pixhawk
-        uses: actions/upload-artifact@v3
-        with:
-          name: pixhawk
-          path: bin/
-          retention-days: 1
-
-  # NOTE: private repo is needed for this build, which is not available for forked PRs
-  saluki-v1_default:
-    name: build saluki-v1_default
-    runs-on: ubuntu-latest
-    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-    steps:
-      - name: Checkout px4-firmware
-        uses: actions/checkout@v3
-        with:
-          token: ${{ secrets.GH_REPO_TOKEN }}
-          submodules: recursive
-          path: px4-firmware
-          fetch-depth: 0
-      - name: Run px4-firmware saluki-v1_default build
-        run: |
-          set -eux
-          mkdir -p bin
-          cd px4-firmware/
-          ./build.sh ../bin/ saluki-v1_default
-          ls ../bin
-      - name: Upload saluki-v1_default
-        uses: actions/upload-artifact@v3
-        with:
-          name: pixhawk
-          path: bin/
-          retention-days: 1
-
-  saluki-v1_bootloader:
-    name: build saluki-v1_bootloader
-    runs-on: ubuntu-latest
-    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'push' || github.event_name == 'workflow_dispatch'
```

Comment:
This if I don't understand. It is a bit complex but it seems to check if this is eighter pull_request, push or workflow_diplatch and only run this step if one of those is true. But this workflow is only triggered for those events. So I didn't include this condition, but maybe there is something I don't see now?